crates/bevy_asset/src/server/mod.rs

```diff
@@ -228,6 +192,22 @@ impl AssetServer {
         Err(MissingAssetLoaderForExtensionError { extensions })
     }
 
+    /// Retrieves the default [`AssetLoader`] for the given [`Asset`] [`TypeId`], if one can be found.
+    pub async fn get_asset_type_id_asset_loader<'a>(
```

Comment:
Help me think: Is this more ergonomic for the user, or could something like

`pub async fn get_asset_asset_loader<'a, A: Asset>(&self) -> Result<...>`

work, where instead of providing the `TypeId` it's retrieved via the info impl'd on `Asset`?

Reply:
Fair point! I've renamed `get_asset_type_id_asset_loader` to `get_asset_loader_with_asset_type_id` and added `get_asset_loader_with_asset_type` which gets the `TypeId` for you. I think having the type-erased option is still important because it's easy to go from concrete -> erased, but not the other way around.